File: My-Resturant-Frontend/src/app/interFaces/Order.ts
import { cartItem } from "./Item"
import { cartMeal } from "./Meals"

export interface order{
 id:number,
 creationDate:Date,
 costumerId:number,
 customerName:string | null,
 orderStatus:string,
 deliveryAdress:string,
 costumerNotes:string|null,
 rating:string|null,
 netPrice:number,
 discountCode:string|null
}
export interface createOrGetOrder{
 costumerId:number,
 orderStatus:string|null,
 deliveryAdress:string,
 costumerNotes:string|null,
 discountCode:string|null,
 rating:string|null,
 netPrice:number,
 cartItems:cartItem[],
 cartMeals:cartMeal[]
}
export interface updateOrder{
  rating:string|null,
  deliveryAdress:string|null,
  customerNotes:string|null,

}

File: My-Resturant-API/Entities/Order.cs
namespace My_Resturant.Entities
{
    public class Order:MainEntity
    {
        public int costumerId { get; set; }
        public int orderStatus {  get; set; }
        public string deliveryAdress { get; set; }
        public string? costumerNotes { get; set; }
        public int? rating { get; set; }
        public double netPrice { get; set; }
        public string? discountCode { get; set; }

    }
}
